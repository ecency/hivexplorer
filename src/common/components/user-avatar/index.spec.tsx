import React from "react";
import TestRenderer from "react-test-renderer";
import { StaticRouter } from "react-router-dom";

import UserAvatar from "./index";

jest.mock("../../constants/defaults.json", () => ({
  imageServer: "https://images.ecency.com"
}));

describe("UserAvatar", () => {
  it("(1) Should render small size", () => {
    const props = { username: "good-karma", size: "small" };
    const renderer = TestRenderer.create(
      <StaticRouter location="/@username" context={{}}>
        <UserAvatar {...props} />
      </StaticRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("(2) Should render normal size", () => {
    const props = { username: "good-karma", size: "normal" };
    const renderer = TestRenderer.create(
      <StaticRouter location="/@username" context={{}}>
        <UserAvatar {...props} />
      </StaticRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("(3) Should render medium size", () => {
    const props = { username: "good-karma", size: "medium" };
    const renderer = TestRenderer.create(
      <StaticRouter location="/@username" context={{}}>
        <UserAvatar {...props} />
      </StaticRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("(4) Should render large size", () => {
    const props = { username: "good-karma", size: "large" };
    const renderer = TestRenderer.create(
      <StaticRouter location="/@username" context={{}}>
        <UserAvatar {...props} />
      </StaticRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("(5) Should render xlarge size", () => {
    const props = { username: "good-karma", size: "xLarge" };
    const renderer = TestRenderer.create(
      <StaticRouter location="/@username" context={{}}>
        <UserAvatar {...props} />
      </StaticRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
