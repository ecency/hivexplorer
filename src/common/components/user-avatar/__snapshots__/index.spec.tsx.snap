// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar (1) Should render small size 1`] = `
Array [
  <span
    className="user-avatar small"
    style={
      Object {
        "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
      }
    }
  />,
  <a
    href="/@good-karma"
    onClick={[Function]}
  >
    good-karma
  </a>,
]
`;

exports[`UserAvatar (2) Should render normal size 1`] = `
Array [
  <span
    className="user-avatar normal"
    style={
      Object {
        "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/small)",
      }
    }
  />,
  <a
    href="/@good-karma"
    onClick={[Function]}
  >
    good-karma
  </a>,
]
`;

exports[`UserAvatar (3) Should render medium size 1`] = `
Array [
  <span
    className="user-avatar medium"
    style={
      Object {
        "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/medium)",
      }
    }
  />,
  <a
    href="/@good-karma"
    onClick={[Function]}
  >
    good-karma
  </a>,
]
`;

exports[`UserAvatar (4) Should render large size 1`] = `
Array [
  <span
    className="user-avatar large"
    style={
      Object {
        "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/medium)",
      }
    }
  />,
  <a
    href="/@good-karma"
    onClick={[Function]}
  >
    good-karma
  </a>,
]
`;

exports[`UserAvatar (5) Should render xlarge size 1`] = `
Array [
  <span
    className="user-avatar xLarge"
    style={
      Object {
        "backgroundImage": "url(https://images.ecency.com/u/good-karma/avatar/large)",
      }
    }
  />,
  <a
    href="/@good-karma"
    onClick={[Function]}
  >
    good-karma
  </a>,
]
`;
